src/main/java/ch/wisv/areafiftylan/teams/controller/TeamRestController.java

```diff
@@ -96,6 +96,42 @@ public TeamRestController(TeamService teamService, SeatService seatService) {
                 "Team successfully created at " + httpHeaders.getLocation(), team);
     }
 
+    /**
+     * The method to handle POST requests on the /teams endpoint. This assigns a new captain to an existing team.
+     * Users can only assign a new captain to a team if they are the captain of that team.
+     * Admins can also assign a new captain to a team.
+     *
+     * @param user The logged in user
+     * @param teamID The ID of the team
+     * @param newCaptainEmail The email of the new captain
+     *
+     * @return Return status message of the operation
+     */
+    @PreAuthorize("isAuthenticated()")
+    @JsonView(View.Public.class)
+    @PostMapping("/{teamID}/changecaptain")
```

Comment:
A better REST style request would be a PUT request on `/{teamid}/captain`. A PUT request indicates a replace of a resource, which is the "captain" in this case